src/libraries/System.Console/src/System/IO/StdInReader.cs

```diff
@@ -308,6 +308,12 @@ internal ConsoleKey GetKeyFromCharValue(char x, out bool isShift, out bool isCtr
             return default(ConsoleKey);
         }
 
+        private bool IsPotentialKeyCharacter(char keyChar)
+        {
+            // symbol or punctuation ascii characters
+            return keyChar < '\x0080' && (char.IsSymbol(keyChar) || char.IsPunctuation(keyChar));
```

Comment:
I was going to suggest `char.IsAscii(char)` but it's private for some reason (don't know why, seems like a reasonable API). There is `Rune.IsAscii(keyChar)` but I think that's less readable 😐

Reply:
Perhaps something we could make public?

Reply:
https://github.com/dotnet/runtime/issues/40936